### ISSUE TITLE
Mobile wallets not working

### ISSUE THREAD

**Issue by aaronbarnardsound** [blocknative/onboard#244]
A bug was introduced in release `1.3.4` with the changes that fixed access to the Opera provider. The provider needs to be called directly as some providers rely on the `this` value to work correctly.

**Comment by aaronbarnardsound** [comment]
Closed with #243

**Comment by aaronbarnardsound** [comment]
This bug was caused due to the Opera provider not having a `sendAsync` function on it (web3.js seems to delete it for some reason). The fix for that was to fallback to the `send` function by assigning it to a variable and then calling that variable with the parameters and callback:

```javascript
const method = provider.sendAsync || provider.send
method(params, callback)
```

This change fixed the Opera bug, but introduced a new bug. Some providers rely on `this` being bound to the provider object and being available. By assigning the method to a variable, `this` is no longer bound to the provider object causing errors in some providers.

The fix in #243 makes sure to check that `provider.sendAsync` is defined and is a function type and then calls it directly rather than assigning it to a variable, otherwise it will then do the same checks and call for `provider.send`.
